grammar.js

```diff
@@ -209,149 +189,262 @@ module.exports = grammar({
-      field("name", choice($.identifier, $.dots))
+      $._parameter_with_default,
+      $._parameter_without_default
+    ),
+
+    _parameter_with_default: $ => seq(
+      field("name", $.identifier), 
+      "=", 
+      optional(field("default", $._expression))
     ),
 
+    _parameter_without_default: $ => field("name", choice($.identifier, $.dots)),
+
     // Control flow.
-    if: $ => prec.right(seq(
+    if_statement: $ => withPrec(PREC.IF, seq(
       "if",
       repeat($._newline),
-      "(",
+      $._open_parenthesis,
       field("condition", $._expression),
-      ")",
+      $._close_parenthesis,
       repeat($._newline),
       field("consequence", $._expression),
       // No `repeat($._newline)` here. Specially handled in the scanner instead.
-      optional(seq(alias($._else, "else"), field("alternative", $._expression)))
+      optional(seq(
+        $._else, 
+        field("alternative", $._expression)
+      ))
     )),
 
-    for: $ => prec.right(seq(
+    for_statement: $ => withPrec(PREC.FUNCTION_OR_LOOP, seq(
       "for",
       repeat($._newline),
-      "(",
+      $._open_parenthesis,
       field("variable", $.identifier),
       "in",
       field("sequence", $._expression),
-      ")",
+      $._close_parenthesis,
       repeat($._newline),
       optional(field("body", $._expression))
     )),
 
-    while: $ => prec.right(seq(
+    while_statement: $ => withPrec(PREC.FUNCTION_OR_LOOP, seq(
       "while",
       repeat($._newline),
-      "(",
+      $._open_parenthesis,
       field("condition", $._expression),
-      ")",
+      $._close_parenthesis,
       repeat($._newline),
       optional(field("body", $._expression))
     )),
 
-    repeat: $ => prec.right(seq(
+    repeat_statement: $ => withPrec(PREC.FUNCTION_OR_LOOP, seq(
       "repeat",
       repeat($._newline),
       optional(field("body", $._expression))
     )),
 
     // Blocks.
-    "{": $ => prec.right(seq(
-      "{",
+    braces: $ => withPrec(PREC.BLOCK, seq(
+      $._open_brace,
       repeat(field("body", choice($._expression, $._semicolon, $._newline))),
-      optional("}")
+      optional($._close_brace)
     )),
 
-    "(": $ => prec.right(seq(
-      "(",
+    parentheses: $ => withPrec(PREC.BLOCK, seq(
+      $._open_parenthesis,
       repeat(field("body", choice($._expression, $._newline))),
-      optional(")")
+      optional($._close_parenthesis)
     )),
 
     // Function calls and subsetting.
-    call: $ => prec.right(16, seq($._expression, field("arguments", alias($._call_arguments,    $.arguments)))),
-    "[":  $ => prec.right(16, seq($._expression, field("arguments", alias($._subset_arguments,  $.arguments)))),
-    "[[": $ => prec.right(16, seq($._expression, field("arguments", alias($._subset2_arguments, $.arguments)))),
-
-    // Dummy rule; used for aliasing so we can easily search the AST.
-    arguments: $ => $._expression,
```

Comment:
Neat side note, we don't actually need this, we can do `alias($.subset_arguments, $.arguments)` in the `subset:` rule instead